cmd/drone-gcr/main.go

```diff
@@ -16,45 +18,80 @@ import (
 	docker "github.com/drone-plugins/drone-docker"
 )
 
-// gcr default username
-var username = "_json_key"
+type Config struct {
+	Repo             string
+	Registry         string
+	Password         string
+	WorkloadIdentity bool
+	Username         string
+	RegistryType     string
+}
 
-func main() {
-	// Load env-file if it exists first
+func loadConfig() Config {
+	// Default username
+	username := "_json_key"
+
+	// Load env-file if it exists
 	if env := os.Getenv("PLUGIN_ENV_FILE"); env != "" {
-		godotenv.Load(env)
+		if err := godotenv.Load(env); err != nil {
+			log.Fatalf("Error loading .env file: %v", err)
+		}
 	}
 
-	var (
-		repo     = getenv("PLUGIN_REPO")
-		registry = getenv("PLUGIN_REGISTRY")
-		password = getenv(
-			"PLUGIN_JSON_KEY",
-			"GCR_JSON_KEY",
-			"GOOGLE_CREDENTIALS",
-			"TOKEN",
-		)
-		workloadIdentity = parseBoolOrDefault(false, getenv("PLUGIN_WORKLOAD_IDENTITY"))
+	location := getenv("PLUGIN_LOCATION")
+	repo := getenv("PLUGIN_REPO")
+
+	password := getenv(
+		"PLUGIN_JSON_KEY",
+		"GCR_JSON_KEY",
+		"GOOGLE_CREDENTIALS",
+		"TOKEN",
 	)
-	// set username and password
+	workloadIdentity := parseBoolOrDefault(false, getenv("PLUGIN_WORKLOAD_IDENTITY"))
 	username, password = setUsernameAndPassword(username, password, workloadIdentity)
-	// default registry value
+
+	registryType := getenv("PLUGIN_REGISTRY_TYPE")
+	if registryType == "" {
+		registryType = "GCR"
+	}
+
+	registry := getenv("PLUGIN_REGISTRY")
 	if registry == "" {
-		registry = "gcr.io"
+		switch registryType {
+		case "GCR":
+			registry = "gcr.io"
+		case "GAR":
+			if location == "" {
+				logrus.Fatalf("Error: For GAR, LOCATION must be set")
```

Comment:
can you change the error message to, 

"Error: For REGISTRY_TYPE of GAR, LOCATION must be set" 

and squash your commits.

Reply:
done and I'll squash on merge :)

Reply:
@eoinmcafee00 you didnt push. no change yet

Reply:
https://github.com/drone-plugins/drone-docker/pull/404/files#diff-c2b522394f8c3e1528f8508b7d9fbb24b34093aca68a56b4cf0bea2b85bf5d98R65